h_malloc.c

```diff
@@ -66,6 +67,10 @@ static atomic_uint thread_arena_counter = 0;
 static const unsigned thread_arena = 0;
 #endif
 
+#ifdef MEMTAG
+bool __is_memtag_enabled = true;
```

Comment:
Stupid question, but an attacker with an arbitrary write could simply overwrite this variable to disable MTE, couldn't she?

Reply:
Yes, but the whole point of MTE is that arbitrary overflows no longer provides a way to overwrite anything without a matching tag. It would still be better to include this in the read-only section and toggle it.